helix-term/tests/test/commands.rs

```diff
@@ -385,3 +385,70 @@ async fn test_character_info() -> anyhow::Result<()> {
 
     Ok(())
 }
+
+#[tokio::test(flavor = "multi_thread")]
+async fn test_delete_char_backward() -> anyhow::Result<()> {
+    // don't panic when deleting overlapping ranges
+    test((
+        platform_line("#(x|)# #[x|]#").as_str(),
```

Comment:
The `as_str` actually shouldn't be necessary any more after 5b07ce7

Reply:
I took the chance to fix this for all tests in that file and therefore made this a separate commit [d605a34](https://github.com/helix-editor/helix/pull/6024/commits/d605a341073325b7f406db524a246e756498c51a)